docs/development.txt

```diff
@@ -0,0 +1,82 @@
+the proposal are made, we wish to minimize the amount of rework required.
+
+[[branching]]
+Branching
+---------
+For features that involve only one developer, developers will work in their own JanusGraph forks, managing their
+own branches, and submitting pull requests when ready. Feature branches in the main JanusGraph repository will be
+created when more than one developer collaborates on a feature.
+
+Branch Naming Conventions
+~~~~~~~~~~~~~~~~~~~~~~~~~
+All branch names shall be prepended with `Issue_#_`.
+
+[[pull-requests]]
+Pull Requests
+-------------
+
+Most changes to JanusGraph must be reviewed before they are committed (RTC). This workflow is performed using Github's
+pull request process. In instances where the committer deems the full process
+unnecessary, commit-then-review (CTR) may be employed. This should be used infrequently and only for extremely trivial
+changes. Changes of this sort may include:
+
+* Documentation typo or small documentation addition
+* Addition of a new test case
+
+Any commit that is made following CTR shall include the fact that it is a CTR in the commit comments.
+
+Users wishing to contribute to JanusGraph should fork the https://github.com/janusgraph/janusgraph[JanusGraph]
+repository and then submit pull requests using the Github pull request process. Please include the relevant issue
+number in the title of your pull request. A vote (-1, 0, +1) is required before a pull request may be merged.
+Non-committers are free, and encouraged to vote. Their votes will be non-binding, but can be taken into consideration
+and are still very valuable community input. The following rules apply to the voting process.
+
+* Three +1 votes are required merge a pull request
+* One or more -1 votes will veto the pull request until the noted issue(s) are addressed and the pull request
+is resubmitted
+* Committers may vote on their own pull requests
+* Pull requests that are limited to one component require at least one of the +1 votes to come from the relevant
+TSC member. If TSC members are unable to review after 2 weeks, the pull request may be merged
+after three +1 votes
+
+[[release-policy]]
+Release Policy
+--------------
+
+Any JanusGraph committer may propose a release. To propose a release, simple start a new [RELEASE] thread on
+https://groups.google.com/forum/#!forum/janusgraph-dev[janusgraph-dev] proposing the new release and requesting feedback on what should be included in the release. After
+consensus is reached, decide on a date to start the release code freeze. Allow for one week between the start of the
+code freeze and the beginning of the release vote. During that week, a release manager shall create release
+artifacts. Committers will be given seven days to review and vote on the release artifacts. Three +1 votes are required for a release to happen.
+If one or more committers casts a -1 vote, the release will be halted.
```

Comment:
In Apache projects, release votes can't be vetoed, they require at least 3 +1 votes and more +1s than -1s. Just pointing that out in case we want to follow suit.